Revamping for easier setup.

hackintoshrao opened the pull request:
- Use docker compose to start Minio distributed server along with the
  worker process.
- Using supervisord to control Minio process.
- The goal was to make the setup, start, control, monitor of fault
  tolerant tests very easy.
- see README.md for more instructions and changes.

harshavardhana commented on Dockerfile:
```diff
@@ -0,0 +1,35 @@
+# supervisord.conf contains configuration for controlling Minio and worker processes.
+COPY supervisord.conf /etc/supervisor/conf.d/supervisord.conf
+
+RUN \
```
Always combine multiple RUNs together in docker, to avoid unnecessary layers being created.

harshavardhana commented on README.md:
````diff
@@ -3,64 +3,115 @@ A framework for testing Minio's fault tolerance capability.
-$ go get -d github.com/minio/chaos` 
-```
+  ```sh
+  $ git clone https://github.com/minio/chaos.git && cd chaos
````
Will it work like this? don't you need to clone this in GOPATH?

hackintoshrao replied:
Since docker compose is used to run the project this is fine.

harshavardhana replied:
If that is the case, we should avoid mentioning these steps.

harshavardhana commented on docker-compose.yml:
```diff
@@ -0,0 +1,42 @@
+    volumes:
+      - ./logs/minio4:/var/log/minio    
+    cap_add:
+      - SYS_ADMIN
```
Remove all unnecessary files in `logs` directory.

harshavardhana commented on worker/chaos-worker.go:
```diff
@@ -109,19 +110,12 @@ func (w *Worker) CheckMinioHealth(args *string, reply *struct{}) error {
 	log.Println("Initializing the Node for the Chaos test.")
 	// Verifies whether Minio is running on the specified port.
-	err := IsMinioRunning(args.Addr)
+	addr := "http://127.0.0.1:9000"
```
why is this hardcoded?

hackintoshrao replied:
Worker will be running on same host as Minio server, so it pings localhost.

harshavardhana replied:
No reason to set the 'addr' then. Move it as argument to 'IsMinioRunning("http://127.0.0.1:9000")` directly. Also what happens if the Minio server is running on non-standard port?. Since this is a controller env i don't expect such things to occur.

hackintoshrao commented:
Addressed all the comments.  

Here is the brief summary about purpose of the revamp,

- Fault tolerant tests are hard to setup, the worker, Minio server, systemd configuration has to deployed on all the nodes, this is a laborious and time consuming process.
- This manual effort involved makes fault tolerant tests makes it less popular and only few would be interested to perform this sort of tests.
- This revamp makes the setup super simple. Its just a matter of `docker-compose up` to setup the environment, and its really easy to investigate the logs and backend. 
- There can be a argument about effectiveness of the test in comparison with running it against manual Minio deployment. Though Minio distributed mode can be run on single host its necessary that we run fault tolerant tests in a multihost environment, Docker's networking isolation between hosts is effective enough to simulate a multi host environment. 
- The tremendous improvement in ease of running test means that its more likely many more people will start testing the fault tolerant capabilities of Minio and it means that there's a greater likelyhood of finding fault tolerant related bugs.

harshavardhana commented on README.md:
````diff
@@ -3,64 +3,115 @@ A framework for testing Minio's fault tolerance capability.
+  $ docker-compose exec minio4 supervisorctl stop minio
+  ```
+
+- Uplaod a file from host.
````
Typo. run `misspell` on the README.md

harshavardhana commented on config.json:
```diff
@@ -0,0 +1,108 @@
+{
+	"version": "16",
```
We have moved to config version 18 the latest release.. perhaps this should reflect that.

harshavardhana commented on master/chaos-worker.go:
```diff
@@ -20,17 +20,13 @@ import (
 	"fmt"
```
Looks like the 'logs' directory is still there, remove `.DS_Store` files.

hackintoshrao commented:
>  Also what happens if the Minio server is running on non-standard port? 

 We are running Minio using compose and we know that Minio is running on port 9000. One needs to edit the compose file to do make Minio run in a different port, since its the container port there's no needs to make Minio run on a different port. The code will fail to apply for a generic case of testing manually, its built for running in a fixed environment of docker compose.
